### DISCUSSION TITLE
firebird connection

### DISCUSSION THREAD

**Discussion by WellingtonBS** [dbeaver/dbeaver#13272] [Q&A]
How to change the dialect of a connection. 

I'm with the error

SQL Error [335544793] [42000]: Dynamic SQL Error; SQL error code = -817; Metadata update statement is not allowed by the current database SQL dialect 1 [SQLState:42000, ISC error code:335544793]

**Comment by uslss** [comment]
probably, you could find more info here https://www.firebirdsql.org/pdfmanual/html/gfix-dialect.html

**Comment by WellingtonBS** [comment]
thank you very much, THE DB is in the correct dialect, I use with other tools, I believe it is some config of DBeaver in the format of connection dialect, has to change the format?

**Reply by isabellamariano** [reply]
Hi Wellington :)
did you find the solution for this? im facing the same issue and using ibexpert i dont have this problem, but in dbeaver i got this issue.

sorry if have typos..

**Comment by tonne1** [comment]
Two ideas which might help:
- update the Firebird JDBC driver to (currently) Jaybird JCA/JDBC driver 5.0.4 java11 RELEASE. The preinstalled drivers are not up-to-date. That resolved connection issues for me, plus the driver supports Firebird 5!

- in Connection Settings, Main tab, Driver Settings button, Advanced parameters tab there is a setting under category DDL "Use legacy SQL dialect for DDL". That should be OFF if you don't need the old dialect
